localcache/localcache.go

```diff
@@ -0,0 +1,119 @@
+// Package localcache provides a simple lru cache mechanism storing data in memory.
+package localcache
+
+import (
+		"container/list"
+		"container/heap"
+		"sync"
+		"time"
+)
+
+// Cache is an interface for cache implementation.
+type Cache interface {
+		Get(key string) (interface{}, bool)
+		Set(key string, value interface{})
+}
+
+// localCache is a simple lru cache implementation with ttl.
+type localCache struct {
+		m        sync.Mutex
+		ttl      time.Duration
+		cap      int
+		size     int
+		store    map[string]*list.Element
+		lrulist  *list.List
+		exppq		 priorityQueue
```

Comment:
It's better to run `linter` before committing.

Reply:
fixed